List is not supported as injected Processor property

I'm using spoon-maven-plugin 2.7.

Despite what is documented in https://github.com/SpoonLabs/spoon-maven-plugin#how-to-pass-properties-to-a-processor there does not seem to be any kind of conversion in the Processor properties system.

For example if I do the following in my processor:

```
    @Property
    private List<String> forbidden;
```

I get:

```
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal fr.inria.gforge.spoon:spoon-maven-plugin:2.7:generate (default) on project xwiki-commons-cache-api: Exception during the spoonify of the target project.
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:213)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:51)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:309)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:194)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:107)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:955)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:290)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:194)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: org.apache.maven.plugin.MojoExecutionException: Exception during the spoonify of the target project.
    at fr.inria.gforge.spoon.Spoon.execute (Spoon.java:201)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:134)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:208)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:51)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:309)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:194)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:107)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:955)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:290)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:194)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: org.apache.maven.plugin.MojoExecutionException: Exception during the spoonify of the target project.
    at fr.inria.gforge.spoon.metrics.PerformanceDecorator.execute (PerformanceDecorator.java:24)
    at fr.inria.gforge.spoon.Spoon.execute (Spoon.java:197)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:134)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:208)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:51)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:309)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:194)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:107)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:955)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:290)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:194)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: spoon.SpoonException: java.lang.IllegalArgumentException: Can not set java.util.List field org.xwiki.tool.spoon.ForbiddenInvocationProcessor.forbidden to java.lang.String
    at spoon.testing.utils.ProcessorUtils.initProperties (ProcessorUtils.java:52)
    at spoon.processing.AbstractProcessor.initProperties (AbstractProcessor.java:128)
    at org.xwiki.tool.spoon.ForbiddenInvocationProcessor.initProperties (ForbiddenInvocationProcessor.java:58)
    at spoon.processing.AbstractProcessor.init (AbstractProcessor.java:117)
    at spoon.support.QueueProcessingManager.process (QueueProcessingManager.java:115)
    at spoon.support.QueueProcessingManager.process (QueueProcessingManager.java:131)
    at spoon.support.compiler.jdt.JDTBasedSpoonCompiler.instantiateAndProcess (JDTBasedSpoonCompiler.java:178)
    at spoon.Launcher.process (Launcher.java:712)
    at spoon.Launcher.run (Launcher.java:658)
    at fr.inria.gforge.spoon.metrics.PerformanceDecorator.execute (PerformanceDecorator.java:22)
    at fr.inria.gforge.spoon.Spoon.execute (Spoon.java:197)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:134)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:208)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:51)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:309)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:194)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:107)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:955)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:290)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:194)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: java.lang.IllegalArgumentException: Can not set java.util.List field org.xwiki.tool.spoon.ForbiddenInvocationProcessor.forbidden to java.lang.String
    at sun.reflect.UnsafeFieldAccessorImpl.throwSetIllegalArgumentException (UnsafeFieldAccessorImpl.java:167)
    at sun.reflect.UnsafeFieldAccessorImpl.throwSetIllegalArgumentException (UnsafeFieldAccessorImpl.java:171)
    at sun.reflect.UnsafeObjectFieldAccessorImpl.set (UnsafeObjectFieldAccessorImpl.java:81)
    at java.lang.reflect.Field.set (Field.java:764)
    at spoon.testing.utils.ProcessorUtils.initProperties (ProcessorUtils.java:50)
    at spoon.processing.AbstractProcessor.initProperties (AbstractProcessor.java:128)
    at org.xwiki.tool.spoon.ForbiddenInvocationProcessor.initProperties (ForbiddenInvocationProcessor.java:58)
    at spoon.processing.AbstractProcessor.init (AbstractProcessor.java:117)
    at spoon.support.QueueProcessingManager.process (QueueProcessingManager.java:115)
    at spoon.support.QueueProcessingManager.process (QueueProcessingManager.java:131)
    at spoon.support.compiler.jdt.JDTBasedSpoonCompiler.instantiateAndProcess (JDTBasedSpoonCompiler.java:178)
    at spoon.Launcher.process (Launcher.java:712)
    at spoon.Launcher.run (Launcher.java:658)
    at fr.inria.gforge.spoon.metrics.PerformanceDecorator.execute (PerformanceDecorator.java:22)
    at fr.inria.gforge.spoon.Spoon.execute (Spoon.java:197)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:134)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:208)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:51)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:309)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:194)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:107)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:955)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:290)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:194)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
```

--- a/pom.xml
+++ b/pom.xml
@@ -240,7 +240,11 @@
       <groupId>org.apache.commons</groupId>
       <artifactId>commons-lang3</artifactId>
       <version>3.5</version>
-      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>com.fasterxml.jackson.core</groupId>
+      <artifactId>jackson-databind</artifactId>
+      <version>2.9.2</version>
     </dependency>
   </dependencies>
 

--- a/src/main/java/spoon/processing/ProcessorPropertiesImpl.java
+++ b/src/main/java/spoon/processing/ProcessorPropertiesImpl.java
@@ -16,15 +16,24 @@
  */
 package spoon.processing;
 
+import org.apache.commons.lang3.ClassUtils;
+
 import java.util.HashMap;
 import java.util.Map;
 
 public class ProcessorPropertiesImpl implements ProcessorProperties {
-
 	private final Map<String, Object> _properties = new HashMap<>();
 
 	public <T> T get(Class<T> type, String name) {
-		return (T) _properties.get(name);
+		if (type.isPrimitive()) {
+			type = (Class<T>) ClassUtils.primitiveToWrapper(type);
+		}
+		T result = (T) _properties.get(name);
+		if (result == null) {
+			return null;
+		} else {
+			return (type.isAssignableFrom(result.getClass())) ? result : null;
+		}
 	}
 
 	public void set(String name, Object o) {

--- a/src/main/java/spoon/testing/utils/ProcessorUtils.java
+++ b/src/main/java/spoon/testing/utils/ProcessorUtils.java
@@ -31,7 +31,7 @@ import java.lang.reflect.Field;
 import java.util.Collection;
 
 public final class ProcessorUtils {
-	private final static ObjectMapper converter = new ObjectMapper();
+	private static final ObjectMapper converter = new ObjectMapper();
 
 	private ProcessorUtils() {
 		throw new AssertionError();
@@ -58,11 +58,11 @@ public final class ProcessorUtils {
 						obj = properties.get(String.class, f.getName());
 						if (obj != null) {
 							try {
-								obj = converter.readValue((String)obj, f.getType());
+								obj = converter.readValue((String) obj, f.getType());
 								f.setAccessible(true);
 								f.set(p, obj);
 							} catch (Exception e) {
-								throw new SpoonException("Error while assigning the value to "+f.getName(), e);
+								throw new SpoonException("Error while assigning the value to " + f.getName(), e);
 							}
 						} else {
 							p.getFactory().getEnvironment().report(p, Level.WARN,

--- a/src/main/java/spoon/testing/utils/ProcessorUtils.java
+++ b/src/main/java/spoon/testing/utils/ProcessorUtils.java
@@ -16,6 +16,7 @@
  */
 package spoon.testing.utils;
 
+import com.fasterxml.jackson.databind.ObjectMapper;
 import org.apache.log4j.Level;
 import spoon.Launcher;
 import spoon.SpoonException;
@@ -30,6 +31,8 @@ import java.lang.reflect.Field;
 import java.util.Collection;
 
 public final class ProcessorUtils {
+	private final static ObjectMapper converter = new ObjectMapper();
+
 	private ProcessorUtils() {
 		throw new AssertionError();
 	}
@@ -52,9 +55,19 @@ public final class ProcessorUtils {
 							throw new SpoonException(e);
 						}
 					} else {
-						p.getFactory().getEnvironment().report(p, Level.WARN,
-								"No value found for property '" + f.getName() + "' in processor " + p.getClass()
-										.getName());
+						obj = properties.get(String.class, f.getName());
+						if (obj != null) {
+							try {
+								obj = converter.readValue((String)obj, f.getType());
+								f.setAccessible(true);
+								f.set(p, obj);
+							} catch (Exception e) {
+								throw new SpoonException("Error while assigning the value to "+f.getName(), e);
+							}
+						} else {
+							p.getFactory().getEnvironment().report(p, Level.WARN,
+									"No value found for property '" + f.getName() + "' in processor " + p.getClass().getName());
+						}
 					}
 				}
 			}

--- a/src/test/java/spoon/MavenLauncherTest.java
+++ b/src/test/java/spoon/MavenLauncherTest.java
@@ -11,7 +11,7 @@ public class MavenLauncherTest {
 		// without the tests
 		MavenLauncher launcher = new MavenLauncher("./", MavenLauncher.SOURCE_TYPE.APP_SOURCE);
 
-		assertEquals(5, launcher.getEnvironment().getSourceClasspath().length);
+		assertEquals(7, launcher.getEnvironment().getSourceClasspath().length);
 		// 52 because of the sub folders of src/main/java
 		assertEquals(52, launcher.getModelBuilder().getInputSources().size());
 

--- a/src/test/java/spoon/test/processing/ProcessingTest.java
+++ b/src/test/java/spoon/test/processing/ProcessingTest.java
@@ -245,10 +245,16 @@ public class ProcessingTest {
 
 		ProcessorProperties props = new ProcessorPropertiesImpl();
 		props.set("aString", "foo");
-		props.set("anInt", "foo");
 		props.set("anObject", "foo");
+		props.set("anInt", "foo");
 
-		ProcessorUtils.initProperties(p, props);
+		try {
+			ProcessorUtils.initProperties(p, props);
+			fail();
+		} catch (SpoonException e) {
+			assertTrue(e.getMessage().contains("anInt"));
+		}
+
 
 		assertEquals("foo", p.aString);
 		assertEquals(0, p.anInt);

--- a/src/test/java/spoon/test/processing/ProcessingTest.java
+++ b/src/test/java/spoon/test/processing/ProcessingTest.java
@@ -23,10 +23,17 @@ import spoon.test.processing.testclasses.CtInterfaceProcessor;
 import spoon.test.processing.testclasses.CtTypeProcessor;
 import spoon.testing.utils.ProcessorUtils;
 
+import java.util.Arrays;
 import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
 
+import static org.junit.Assert.assertArrayEquals;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
 import static org.junit.Assert.assertSame;
 import static org.junit.Assert.assertTrue;
 import static org.junit.Assert.fail;
@@ -165,6 +172,18 @@ public class ProcessingTest {
 			@Property
 			Object anObject;
 
+			@Property
+			int[] arrayInt;
+
+			@Property
+			List<String> listString;
+
+			@Property
+			boolean[] arrayBoolean;
+
+			@Property
+			Map<String,Double> mapStringDouble;
+
 			@Override
 			public void process() {
 
@@ -172,6 +191,8 @@ public class ProcessingTest {
 		};
 
 		AProcessor p = new AProcessor();
+		Launcher launcher = new Launcher();
+		p.setFactory(launcher.getFactory());
 
 		ProcessorProperties props = new ProcessorPropertiesImpl();
 		props.set("aString", "foo");
@@ -179,11 +200,117 @@ public class ProcessingTest {
 		Object o = new Object();
 		props.set("anObject", o);
 
+		int[] arrayInt = new int[]{ 1, 2, 3};
+		props.set("arrayInt", arrayInt);
+		props.set("listString", Arrays.asList(new String[]{"42"}));
+
+		boolean[] arrayBoolean = new boolean[] {true};
+		props.set("arrayBoolean", arrayBoolean);
+		HashMap<String,Double> mapTest = new HashMap<>();
+		mapTest.put("foobar",42.42);
+		props.set("mapStringDouble", mapTest);
+
 		ProcessorUtils.initProperties(p, props);
 
 		assertEquals("foo", p.aString);
 		assertEquals(5, p.anInt);
 		assertSame(o, p.anObject);
+		assertSame(arrayInt, p.arrayInt);
+		assertEquals(Arrays.asList(new String[]{"42"}), p.listString);
+		assertSame(arrayBoolean, p.arrayBoolean);
+		assertSame(mapTest, p.mapStringDouble);
+	}
+
+	@Test
+	public void testInitPropertiesWithWrongType() throws Exception {
+		class AProcessor extends AbstractManualProcessor {
+			@Property
+			String aString;
+
+			@Property
+			int anInt;
+
+			@Property
+			Object anObject;
+
+			@Override
+			public void process() {
+
+			}
+		};
+
+		AProcessor p = new AProcessor();
+		Launcher launcher = new Launcher();
+		p.setFactory(launcher.getFactory());
+
+		ProcessorProperties props = new ProcessorPropertiesImpl();
+		props.set("aString", "foo");
+		props.set("anInt", "foo");
+		props.set("anObject", "foo");
+
+		ProcessorUtils.initProperties(p, props);
+
+		assertEquals("foo", p.aString);
+		assertEquals(0, p.anInt);
+		assertNull(p.anObject);
+	}
+
+	@Test
+	public void testInitPropertiesWithStringType() throws Exception {
+		class AProcessor extends AbstractManualProcessor {
+			@Property
+			String aString;
+
+			@Property
+			int anInt;
+
+			@Property
+			Object anObject;
+
+			@Property
+			int[] arrayInt;
+
+			@Property
+			List<String> listString;
+
+			@Property
+			boolean[] arrayBoolean;
+
+			@Property
+			Map<String,Double> mapStringDouble;
+
+			@Override
+			public void process() {
+
+			}
+		};
+
+		AProcessor p = new AProcessor();
+		Launcher launcher = new Launcher();
+		p.setFactory(launcher.getFactory());
+
+		ProcessorProperties props = new ProcessorPropertiesImpl();
+		props.set("aString", "foo");
+		props.set("anInt", "42");
+		props.set("anObject", "{}");
+		props.set("arrayInt", "[42,43]");
+		props.set("listString", "[\"foo\", \"bar\"]");
+		props.set("arrayBoolean", "[true]");
+		props.set("mapStringDouble","{\"foo\": 10.21, \"bar\": 14.42}");
+
+		ProcessorUtils.initProperties(p, props);
+
+		assertEquals("foo", p.aString);
+		assertEquals(42, p.anInt);
+		assertNotNull(p.anObject);
+
+		assertArrayEquals(new int[] {42, 43}, p.arrayInt);
+		assertEquals(Arrays.asList(new String[]{"foo", "bar"}), p.listString);
+		assertArrayEquals(new boolean[]{true}, p.arrayBoolean);
+		Map<String, Double> mamap = new HashMap<>();
+		mamap.put("foo", 10.21);
+		mamap.put("bar", 14.42);
+		assertEquals(mamap, p.mapStringDouble);
 	}
 
 	@Test

